docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -230,19 +247,76 @@ void TemplateInferRequest::waitPipeline() {
 void TemplateInferRequest::inferPostprocess() {
     OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, _profilingTask[Postprocess]);
     auto start = Time::now();
-    for (auto&& output : _outputs) {
-        auto outputBlob = output.second;
-        auto networkOutput = _networkOutputBlobs[output.first];
+    for (auto&& networkOutput : _networkOutputs) {
+        {
+            // Touch blob to allocate it
+            Blob::Ptr blob;
+            GetBlob(networkOutput.first.c_str(), blob);
+        }
+        auto outputBlob = _outputs.at(networkOutput.first);
+        auto networkOutputBlob = _networkOutputBlobs[networkOutput.first];
         // perform precision conversion of network output's precision and computational
         // graph output's precision are different
-        if (outputBlob->getTensorDesc().getPrecision() != networkOutput->getTensorDesc().getPrecision()) {
-            blobCopy(networkOutput, outputBlob);
+        if (outputBlob->getTensorDesc().getPrecision() != networkOutputBlob->getTensorDesc().getPrecision()) {
+            blobCopy(networkOutputBlob, outputBlob);
         }
     }
     _durations[Postprocess] = Time::now() - start;
 }
 // ! [infer_request:infer_postprocess]
 
+void TemplateInferRequest::GetBlob(const char* name, Blob::Ptr& data) {
```

Comment:
I think we need to provide an adequate base implementation and not override such generic method.